packages/vats/src/transfer.js

```diff
@@ -0,0 +1,186 @@
+// @ts-check
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+import { coerceToData, dataToBase64 } from '@agoric/network';
+import { AppI } from './bridge-target.js';
+
+const { Fail } = assert;
+
+const ReactionGuard = M.call(M.any()).optional(M.any()).returns(M.any());
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {import('@agoric/vow').VowTools} vowTools
+ */
+export const prepareTransferInterceptor = (zone, vowTools) => {
+  const { watch } = vowTools;
+  const makeTransferInterceptorKit = zone.exoClassKit(
+    'TransferInterceptorKit',
+    {
+      public: AppI,
+      encodeAckWatcher: M.interface('EncodeAckWatcher', {
+        onFulfilled: ReactionGuard,
+      }),
+      sendErrorWatcher: M.interface('SendErrorWatcher', {
+        onRejected: ReactionGuard,
+      }),
+      logErrorWatcher: M.interface('LogErrorWatcher', {
+        onRejected: ReactionGuard,
+      }),
+    },
+    /**
+     * @param {ERef<import('./bridge-target.js').System>} system
+     * @param {ERef<import('./bridge-target.js').App>} tap
+     * @param {boolean} [isActiveTap] Whether the tap is active (can modify
+     *   acknowledgements), or passive (can't cause delays in the middleware).
+     */
+    (system, tap, isActiveTap = false) => ({ isActiveTap, system, tap }),
+    {
+      public: {
+        async upcall(obj) {
+          const { isActiveTap, system, tap } = this.state;
+
+          obj.type === 'VTRANSFER_IBC_EVENT' ||
```

Comment:
I'm seeing this string a lot, it should probably be defined once in internal and used by reference everywhere else.